ssstar/src/objstore/s3.rs

```diff
@@ -1502,130 +1531,23 @@ async fn make_s3_client(
     let aws_config = aws_config_builder.load().await;
 
     let mut s3_config_builder = aws_sdk_s3::config::Builder::from(&aws_config)
-        .app_name(aws_config::AppName::new(APP_NAME).expect("BUG: hard-coded app name is invalid"));
+        .app_name(aws_config::AppName::new(APP_NAME).expect("BUG: hard-coded app name is invalid"))
+        .interceptor(UserAgentInterceptor {
+            user_agent: config
+                .user_agent
+                .clone()
```

Comment:
Use `as_ref()` to convert `Option<T> -> Option<&T>`
```suggestion
                .as_ref()
```